text/trace/0189-instrumentation-layers-suppression.md

```diff
@@ -0,0 +1,180 @@
+# Instrumentation Layers and Suppression
+
+This document describes approach for tracing instrumentation layers, suppressing duplicate layers and unambiguously enriching spans.

```

Comment:
I think it's worth mentioning that the same problem will exists for metrics, even more aggravated (just thinking about a metric "Request Duration" in nested HTTP client spans).

The solution proposed in this document could also be used to suppress those "duplicate" metrics.

Reply:
I suggest keeping it separate for now  - we need to have some clarity around metrics instrumentation - if it's the responsibility of every library or a central component before we can tackle metrics suppression.

Reply:
Agreed. I just think we shouldn't have two different suppression mechanisms for metrics and traces, if possible.